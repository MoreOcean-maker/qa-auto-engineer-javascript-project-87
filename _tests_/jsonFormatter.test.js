import jsonFormatter from '../formatters/json.js';

describe('json formatter', () => {
  test('correctly formats diff to JSON', () => {
    const diff = [
      {
        key: 'timeout',
        type: 'updated',
        oldValue: 50,
        value: 20,
      },
      {
        key: 'verbose',
        type: 'added',
        value: true,
      },
    ];

    const result = jsonFormatter(diff);
    expect(() => JSON.parse(result)).not.toThrow();
    expect(result).toMatchSnapshot();
  });
});
