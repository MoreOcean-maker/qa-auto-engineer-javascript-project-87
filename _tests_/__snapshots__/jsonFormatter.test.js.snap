// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json formatter correctly formats diff to JSON 1`] = `
"[
  {
    "key": "timeout",
    "type": "updated",
    "oldValue": 50,
    "value": 20
  },
  {
    "key": "verbose",
    "type": "added",
    "value": true
  }
]"
`;
